drivers/rn2xx3/rn2xx3_internal.c

```diff
@@ -214,30 +217,11 @@ int rn2xx3_cmd_finalize(rn2xx3_t *dev)
 
 void rn2xx3_mac_tx_start(rn2xx3_t *dev)
 {
-    memset(dev->cmd_buf, 0, strlen(dev->cmd_buf));
-    snprintf(dev->cmd_buf, 7, "mac tx");
-    strcat(dev->cmd_buf, " ");
-    switch (dev->loramac.tx_mode) {
-        case LORAMAC_TX_CNF:
-            strcat(dev->cmd_buf, "cnf");
-            break;
-
-        case LORAMAC_TX_UNCNF:
-            strcat(dev->cmd_buf, "uncnf");
-            break;
-
-        default:
-            DEBUG("[rn2xx3] unsupported join procedure mode\n");
-            break;
-    }
-
-    strcat(dev->cmd_buf, " ");
-
-    char str_port[4];
-    snprintf(str_port, 3, "%d", dev->loramac.tx_port);
-
-    strcat(dev->cmd_buf, str_port);
-    strcat(dev->cmd_buf, " ");
+    size_t p = 0;
+    dev->cmd_buf[p] = 0;
```

Comment:
This cmd_buf[p] = 0 must be after snprintf. Before snprintf it is useless code. (See other remark about this.)